openapi/src/parts/controlled_azure_disk.yaml

```diff
@@ -24,6 +24,49 @@ paths:
         '500':
           $ref: '#/components/responses/ServerError'
 
+  /api/workspaces/v2/{workspaceId}/resources/controlled/azure/disks:
+    parameters:
+      - $ref: '#/components/parameters/WorkspaceId'
+    post:
+      summary: Asynchronously create a new controlled Azure Disk
+      operationId: createAzureDiskV2
+      tags: [ ControlledAzureResource ]
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateControlledAzureDiskRequestV2Body'
+      responses:
+        '200':
+          description: Response to create controlled Azure disk
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CreateControlledAzureResourceResult'
```

Comment:
Question here, currently the createDisk request to WSM returns the wsmResourceId in the response, I might be missing something, but I don't see that supplied in the v2 create or create-result

Reply:
We can get that ID from getting the VM information, but that doesn't work for apps...

Reply:
There was this [discussion](https://github.com/DataBiosphere/terra-workspace-manager/pull/1629/files/67e624e81cadc5a975c04aab7661f1047a6d5d82#diff-663a06eccffd61fc3303d37e8aa796c277376804a18f2cfa765d78db683cd734)

@jdcanas maybe the resource is needed after all? I am not sure about leo's use, but if it is I suggest mimic-ing the rest of the resource result endpoints and leaving the refactoring of the responses to exclude resources to a later effort as there are likely other consumer use cases we need to consider further before shifting to this new pattern.

Reply:
The thing is, we don't need the whole resource in most cases in Leo, so could we just add the resource ID to the `CreateControlledAzureResourceResult`? Then when we do need more info, we can `get` it.

Reply:
Ah, good catch Lucy, missed this requirement for downstream consumption by Leo. Should be able to add `resourceId` to `CreateControlledAzureResourceResult` with no problems as I see it and accomplish both goals of it being generic and satisfying Leo's use case. Will add this and update tests to verify the field's presence.